src/main/java/it/smartcommunitylab/aac/webauthn/auth/WebAuthnRpRegistrationRepository.java

```diff
@@ -0,0 +1,67 @@
+package it.smartcommunitylab.aac.webauthn.auth;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.yubico.webauthn.RelyingParty;
+import com.yubico.webauthn.data.RelyingPartyIdentity;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredentialsRepository;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccountRepository;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnYubicoCredentialsRepository;
+import it.smartcommunitylab.aac.webauthn.provider.WebAuthnIdentityProviderConfigMap;
+
+@Service
+public class WebAuthnRpRegistrationRepository {
+    @Autowired
+    WebAuthnUserAccountRepository userAccountRepository;
+    @Autowired
+    WebAuthnCredentialsRepository webAuthnCredentialsRepository;
+
+    private final Map<String, RelyingParty> rps;
+
+    public WebAuthnRpRegistrationRepository() {
+        this.rps = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Returns the relying party associated with this provider id, if it has been
+     * previously added.
+     */
+    public Optional<RelyingParty> getRpByProviderId(String providerId) {
+        return Optional.ofNullable(rps.get(providerId));
+    }
+
+    public RelyingParty addRp(String providerId, WebAuthnIdentityProviderConfigMap config) {
+        final Pattern localhostPattern = Pattern.compile("^(localhost|127\\.0\\.0\\.1|0\\.0\\.0\\.0)$");
```

Comment:
what is the purpose of checking for localhost? Why use a regex?

Reply:
WebAuthn mandates the use of https when communicating with the server, except when such server is on localhost.
I added these lines in order to enable the use of http instead of https during development

Reply:
Using a regex was just the first way I thought about that. Perhaps we may use another method, like having a pre-defined collection of hostnames/addresses.
```java
final List<String> allowed = Arrays.asList( "localhost", "127.0.0.1", "0.0.0.0" );
final String rpid = config.getRpid();
final boolean isRpidLocalhost = allowed.contains(rpid);
```

Reply:
Done in 9a53d7b7d145fe5eb8309cca2f9fe171746b8a90